libraries/rtos/rtx/TARGET_CORTEX_A/rt_CMSIS.c

```diff
@@ -905,7 +905,7 @@ uint8_t osThreadGetState (osThreadId thread_id) {
   if (__exceptional_mode()) return osErrorISR;     // Not allowed in ISR
 
   ptcb = rt_tid2ptcb(thread_id);                // Get TCB pointer
-  if (ptcb == NULL) return osErrorParameter;
+  if (ptcb == NULL) return INACTIVE;
```

Comment:
Ideally, this function should only be returning the state definitions in [rt_Task.h](https://github.com/mbedmicro/mbed/blob/master/libraries/rtos/rtx/TARGET_CORTEX_M/rt_Task.h), and per the documentation in [Thread.h](https://github.com/mbedmicro/mbed/blob/master/libraries/rtos/rtos/Thread.h), `INACTIVE` is the expected state for threads that have been terminated or otherwise don't exist. However, since there's no state definition for the ISR error I left that line alone.